api/client/application/client.go

```diff
@@ -1046,6 +1047,8 @@ type PendingResourceUpload struct {
 	PendingID string
 	// Type of the resource, a string matching one of the resource.Type
 	Type string
+
+	Res charmresource.Resource
```

Comment:
Why add charmresource.Resource here? It appears that only Name is used from it, which is already in the PendingResourceUpload. As well, PendingResourceUpload is a subset of the charmresource.Resource data as it is.